scripts/eachdist.py

```diff
@@ -573,18 +577,64 @@ def update_version_files(targets, version, packages):
         f'__version__ = "{version}"',
     )
 
+def update_dependency_requirements(deps, pkg, version):
+    updated_deps = []
+    for dep in deps:
+        req = Requirement(dep)
+        if req.name == basename(pkg):
+            specifiers_set = []
+            for spec in req.specifier:
+                if ".dev" in spec.version:
+                    if spec.operator == "<":
+                        specifiers_set.append(Specifier(f"<{spec.version.replace('.dev', '')}"))
+                    elif spec.operator == "<=":
+                        specifiers_set.append(Specifier(f"<={spec.version.replace('.dev', '')}"))
+                    elif spec.operator == "==":
+                        specifiers_set.append(Specifier(f"=={version}"))
+                    elif spec.operator == ">=":
+                        specifiers_set.append(Specifier(f">={spec.version.replace('.dev', '')}"))
+                    elif spec.operator == ">":
+                        specifiers_set.append(Specifier(f">{spec.version.replace('.dev', '')}"))
+                    elif spec.operator == "!=":
+                        specifiers_set.append(Specifier(f"!={spec.version.replace('.dev', '')}"))
+                    elif spec.operator == "~=":
+                        specifiers_set.append(Specifier(f"!={spec.version.replace('.dev', '')}"))
```

Comment:
If not making the change above, there seems to be a typo here, `!=` is used instead of `~=`